File: 0-rabbitmq-fundamentals/1-java/src/main/java/com/starci/rabbitmq/Topology.java
package com.starci.rabbitmq;

/**
 * Shared AMQP topology constants for the producer and every consumer.
 * Keeping names in one place ensures all processes declare the same exchanges,
 * queues, and binding keys — a mismatch would silently drop messages.
 */
public final class Topology {

    /** Utility class — not instantiable. */
    private Topology() {}

    /** Direct exchange: routes by an exact routing-key match (1:1 routing). */
    public static final String EXCHANGE_DIRECT = "orders.direct";
    /** Fanout exchange: ignores the routing key and broadcasts to every bound queue. */
    public static final String EXCHANGE_FANOUT = "events.fanout";
    /** Topic exchange: routes by wildcard pattern (* = one token, # = zero or more). */
    public static final String EXCHANGE_TOPIC = "logs.topic";

    /** Durable queue bound to the direct exchange for the direct consumer. */
    public static final String QUEUE_DIRECT = "orders.created.q";
    /** Durable queue for fanout consumer instance A — each instance needs its own queue. */
    public static final String QUEUE_FANOUT_A = "events.fanout.q.a";
    /** Durable queue for fanout consumer instance B. */
    public static final String QUEUE_FANOUT_B = "events.fanout.q.b";
    /** Durable queue bound to the topic exchange with TOPIC_PATTERN. */
    public static final String QUEUE_TOPIC = "logs.topic.q";

    /** Exact binding key for the direct queue; any other key is dropped by the broker. */
    public static final String DIRECT_ROUTING_KEY = "orders.created";
    /** Topic pattern: # matches zero or more dot-separated tokens. */
    public static final String TOPIC_PATTERN = "logs.#";
}

File: 0-rabbitmq-fundamentals/2-csharp/Topology.cs
namespace RabbitMqFundamentals;

/// <summary>
/// Shared AMQP topology constants for the producer and every consumer.
/// Keeping names in one place ensures all processes declare the same exchanges,
/// queues, and binding keys — a mismatch would silently drop messages.
/// </summary>
public static class Topology
{
    /// <summary>Direct exchange: routes by an exact routing-key match (1:1 routing).</summary>
    public const string ExchangeDirect = "orders.direct";

    /// <summary>Fanout exchange: ignores the routing key and broadcasts to every bound queue.</summary>
    public const string ExchangeFanout = "events.fanout";

    /// <summary>Topic exchange: routes by wildcard pattern (* = one token, # = zero or more).</summary>
    public const string ExchangeTopic = "logs.topic";

    /// <summary>Durable queue bound to the direct exchange for the direct consumer.</summary>
    public const string QueueDirect = "orders.created.q";

    /// <summary>Durable queue for fanout consumer instance A — each instance needs its own queue.</summary>
    public const string QueueFanoutA = "events.fanout.q.a";

    /// <summary>Durable queue for fanout consumer instance B.</summary>
    public const string QueueFanoutB = "events.fanout.q.b";

    /// <summary>Durable queue bound to the topic exchange with <see cref="TopicPattern"/>.</summary>
    public const string QueueTopic = "logs.topic.q";

    /// <summary>Exact binding key for the direct queue; any other key is dropped by the broker.</summary>
    public const string DirectRoutingKey = "orders.created";

    /// <summary>Topic pattern: # matches zero or more dot-separated tokens.</summary>
    public const string TopicPattern = "logs.#";
}
